test/unit/test_sap_cli_checkin.py

```
@@ -1147,6 +1190,76 @@ def test_checkin_fugr_abapgit_all_parameters(self):
+    def test_check_before_save_true_runs_pre_check(self):
+        adt_object = MagicMock()
+        editor = MagicMock()
+        editor.__enter__.return_value = editor
+        adt_object.open_editor.return_value = editor
+
+        bad_result = SimpleNamespace(has_errors=True, messages=iter([]))
+
+        with patch('sap.adt.checks.run_object_check', return_value=bad_result):
+            with self.assertRaises(sap.adt.checks.ObjectCheckFindings) as cm:
+                sap.cli.checkin._write_source_file('CODE', adt_object,
+                                                    source_label='src/foo.clas.abap',
```

Flake8 (7.3.0)

[error] continuation line over-indented for visual indent

(E127)